config/cache/v1alpha1/cache.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package gingersnap.config.cache.v1alpha1;
+import "apimachinery/pkg/api/resource/quantity.proto";
+
+// TODO  option java_package = define package naming for java
+// TODO  option java_outer_classname = define outer class name
+option java_multiple_files = true;
+// TODO option go_package = verify below value is ok
+option go_package = "api/v1alpha1";
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes the desired configuration for a Cache. Only DB Cache Service is supported atm
+message CacheSpec {
+  // Resource profile for the cache provider
+  ResourcesProfile resources = 1;
+  // DatasourceRef or a ServiceBindingRef (TODO clarify)
+  NamespacedRef dataSourceRef = 2;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  DBType dbType = 1;
+  // Additional properties. DB specific
+  map<string,string> connectionProperties = 2;
+  // Reference to the ServiceBinding CR
+  NamespacedRef serviceBindingRef = 3;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message EagerCachingRuleSpec {
+  // Reference to the related Cache CR
+  NamespacedRef  cacheRef = 1;
```

Comment:
Just a general point about naming, I think we should avoid the "Ref" suffix as it just makes the CR more verbose. For example:

```yaml
spec:
  cache:
    name: some-cache
    namespace: some-namespace
```

vs

```yaml
spec:
  cacheRef:
    name: some-cache
    namespace: some-namespace
```

Reply:
I'm referring to this [guideline doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field)

Reply:
Isn't that specifically related to a `ObjectReference`? Looking at the usage in `k8s.io/api/core/v1/types.go` it seems like the `Ref` suffix is only added when the ObjectReference struct is.

Whereas, other resources in the core api that refer to specific instances don't use the suffix. For example, `SecretVolumeSource` and `ConfigMapVolumeSource`:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  containers:
  - name: mypod
    image: redis
    volumeMounts:
    - name: foo
      mountPath: "/etc/foo"
      readOnly: true
  volumes:
  - name: foo
    secret:
      secretName: mysecret
      items:
      - key: username
        path: my-group/my-username
```

Reply:
it's also used sometime around, i.e. there a secretRef [here](https://github.com/kubernetes/api/blob/8384a1b0af6a34368da256be0b98035e4f272024/core/v1/types.go#L840) for a field of type [SecretReference](https://github.com/kubernetes/api/blob/8384a1b0af6a34368da256be0b98035e4f272024/core/v1/types.go#L929)
which is a name:namespace strings pair. Also used with LocalObjectReference which is just a string name
imo a Ref or Name suffix makes clear its a reference

Reply:
Good counter examples :slightly_smiling_face: , let's stick with `Ref` suffix then :+1: